msteams-platform/concepts/deploy-and-publish/appsource/prepare/frequently-failed-cases.md

```diff
@@ -1,246 +1,322 @@
-See the following section for welcome message requirements:
+Search-based messaging extensions must provide text that helps users effectively search.
 
-> [!Note]
-> A welcome message is optional for a channel bot.
+For more information, see the [design best practices for search commands](~/messaging-extensions/design/messaging-extension-design.md#searches).
 
-### Welcome message requirements
+## 8.0 Task module design
 
-* Include a value proposition with the welcome tour.
-* Provide way-forward guidance for using the app.
-* Include guidance on how to sign-up and configure your app.
-* Present easy-to-read text and straightforward dialogue — preferably a card with an actionable welcome tour button that loads a task module.
-* Keep it simple and usable with buttons and cards — avoid long text, chatty dialogue.
-* Include adaptive cards and buttons to make the welcome message more usable.
-* Invoke the welcome message  with one ping, not two or more simultaneous pings.
-* A welcome message must only be shown to the user who configured the app, preferably in a 1:1 personal chat.
-* Personal apps must always provide a welcome message to a user.
-* Never send a personal chat to every member of the team; it is considered spam.
-* Never send the welcome message more than once. Repeating the same welcome message over regular intervals is not allowed and is considered spamming.
+A task module must include an icon and short name of the app in which it is associated.
 
-#### Avoid welcome message spamming
+See comprehensive [task module design guidelines](~/task-modules-and-cards/task-modules/design-teams-task-modules.md).
 
-* **Channel message by bot**. Don't spam users by creating separate new chat posts. Create a single thread post with replies in the same thread.
-* **Personal chat by bot**. Don't send multiple messages. Send one message with complete information. Repeating the same welcome message over regular intervals is not allowed and is considered spamming.
+## 9.0 Meeting extension design
 
-#### Notification-only bot welcome messages
+See comprehensive [meeting extension design guidelines](~/apps-in-teams-meetings/design/designing-apps-in-meetings.md).
 
-Notification-only bots must send a welcome message that includes a message conveying, *"I am a notification-only bot and will not be able to reply to your chats"*.
+### 9.1 Pre- and post-meeting experience
 
-#### Welcome messages in the personal scope
+* Tabs must adhere to general [tab design guidelines](~/tabs/design/tabs.md).
+* Tabs must not have horizontal scrolling.
+* Tabs should have an organized layout when displaying multiple items (for instance, more than 10 polls or surveys). [See an example layout](~/apps-in-teams-meetings/design/designing-apps-in-meetings.md#after-a-meeting).
```

Comment:
Made the reference to "polls or survey" as just a couple examples. Does this work?

Reply:
Yes